### DISCUSSION TITLE
Are there stable releases?

### DISCUSSION THREAD

**Discussion by victorsevero** [grumpycoders/pcsx-redux#1215] [Q&A] [answered]
The official download page only contain nightly releases afaik. Are there stable releases? If yes, where can I find them?

**Comment by nicolasnoble** [comment, marked as answer]
Nope, I'm still not comfortable calling any tagged version "stable" yet. One day.

**Reply by MetalMaxMX** [reply]
So, it's been almost two years since this post. Is it now in better shape for a tagged release? Or still not comfortable?

**Reply by nicolasnoble** [reply]
So much to do, so little time.

Basically, the big issue to call a stable release would be when I'm sure the savestates won't be affected by any further changes. The version number got bumped several times already, invalidating old ones in the process, which is never pleasant for the end users.

https://github.com/grumpycoders/pcsx-redux/blob/main/src/core/sstate.cc#L185-L186

To give a bit of an idea about the process and dependencies, the next big one that'll break savestates is the cd-rom code rewrite, which I'm still due to finish, but life got in the way. The next two others after the cd-rom will be:

- SPU rewrite, for proper synchronization emulation
- More accurate GPU emulation, with delay approximations

The big one which I'm still not sure if it'll change savestates yet is the timers rewrite. After these 3 are done, the savestate format shouldn't change much more anymore, and we could start calling it stable as a result.
